app/jobs/scheduled/problem_checks.rb

```diff
@@ -14,7 +14,12 @@ def execute(_args)
       # not be re-added by the relevant checker, and will be cleared.
       AdminDashboardData.clear_found_scheduled_check_problems
 
-      ::ProblemCheck.scheduled.each do |check|
+      scheduled_checks =
+        ProblemCheckTracker.all.filter_map do |tracker|
+          tracker.check if tracker.check.scheduled? && tracker.ready_to_run?
+        end
```

Comment:
This line in particular made me think `ProblemCheckTracker` and `ProblemCheck` should just be a single `ProblemCheck` DB backed model.